### ISSUE TITLE
[HTML writer] "aria-hidden" and "role" are not valid XHTML 1.0 Transitional attributes

### ISSUE THREAD

**Issue by stroobandt** [jgm/pandoc#8241]
## Tested Pandoc version
pandoc 2.19.1 (2022-08-18)
on Ubuntu 20.04 focal

## Problem
Documents of the type [`XHTML 1.0 Transitional`](https://www.w3.org/TR/xhtml1/) written by the Pandoc [HTML writer](https://pandoc.org/MANUAL.html#general-options) (`--to=html4`) [fail validation](https://validator.w3.org/check?uri=https%3A%2F%2Fhamwaves.com%2Fzfs%2Fen%2F) because of the presence of `aria-hidden="true"` and `role="doc-biblioref"` attributes.

### Invalid aria-hidden attributes
This bug most probably got introduced with [pandoc-2.10 (2020-06-29)](https://pandoc.org/releases.html#pandoc-2.10-2020-06-29).

> - Use latest skylighting. This fixes a bug with lua multiline comments (and may improve handling of other syntaxes as well). IT also adds aria-hidden="true" to the empty a elements, which helps people who use screen readers.

### Invalid role attributes
This bug most probably got introduced with [pandoc 2.7 (2019-03-03)](https://pandoc.org/releases.html#pandoc-2.7-2019-03-03).

> - Implement WAI-ARIA roles for (end)notes, citations, and bibliography (#4213). Note that doc-biblioref is only used when link-citations produces links, since it belongs on links.

## Example
Create an `input.md` Markdown document with below elements and then run:

```bash
pandoc input.md --to=html4 --standalone --citeproc --output=output.html
```

### YAML header
```yaml
---
title: '[HTML writer] "aria-hidden" and "role" are not valid XHTML 1.0 Transitional attributes'
link-citations: true
references:
- id: W3C
  type: article-journal
  title: 'XHTML™ 1.0 The Extensible HyperText Markup Language (Second Edition)'
  author: W3C
  issued:
    year: 2002
    month: 8
    day: 1
  URL: https://www.w3.org/TR/xhtml1/
...
```

### Code block

    ```bash
    echo Code blocks create the invalid aria-hidden attribute.
    ```
Output with aria-hidden:

```html
<div class="sourceCode" id="cb1"><pre class="sourceCode bash"><code class="sourceCode bash">
<span id="cb1-1">
<a href="#cb1-1" aria-hidden="true" tabindex="-1"></a>
echo Code blocks create the invalid aria-hidden attribute.
</span>
</code></pre></div>
```

### Citation

    XHTML 1.0 Transitional[@W3C] with `link-citations: true` in the YAML document header

Output with role:

```html
<p>XHTML 1.0 Transitional<span class="citation">(<a href="#ref-W3C"
role="doc-biblioref">W3C 2002</a>)</span> with
```

## Suggestion
It would be nice to test future HTML writer pull requests in an automated fashion against the [W3C Markup Validation Service](https://validator.w3.org/), which happens to have a very straightforward API.

**Comment by jgm** [comment]
To fix the aria-hidden, we'd need an API change in jgm/skylighting-format-blaze-html, so that its inclusion is configurable.
Feel free to open an issue if you like.
In any case, this should only occur when `numberLines` is used.

I can fix the other issue in pandoc itself.
